jgrapht-core/src/test/java/org/jgrapht/perf/spanning/MinimumSpanningTreePerformanceTest.java

```diff
@@ -0,0 +1,192 @@
+
+        abstract MinimumSpanningTreeAlgorithm<DefaultWeightedEdge> createSolver(
+            Graph<Integer, DefaultWeightedEdge> graph);
+
+        public void setup()
+        {
+            if (generator == null) {
+                // lazily construct generator
+                generator = new GnpRandomGraphGenerator<>(
+                    PERF_BENCHMARK_VERTICES_COUNT, PERF_BENCHMARK_EDGES_PROP, rng, false);
+            }
+
+            DirectedWeightedPseudograph<Integer, DefaultWeightedEdge> weightedGraph =
+                new DirectedWeightedPseudograph<>(DefaultWeightedEdge.class);
+            this.graph = weightedGraph;
+
+            generator.generateGraph(graph, new IntegerVertexFactory(), null);
+
+            for (DefaultWeightedEdge e : weightedGraph.edgeSet()) {
+                weightedGraph.setEdgeWeight(e, rng.nextDouble());
+            }
+        }
+
+        public void run()
+        {
+            MinimumSpanningTreeAlgorithm<DefaultWeightedEdge> algo = createSolver(graph);
+            algo.getSpanningTree();
+        }
+    }
+
+    public static class PrimBenchmark
+        extends BenchmarkBase
+    {
+        @Override
+        MinimumSpanningTreeAlgorithm<DefaultWeightedEdge> createSolver(
+            Graph<Integer, DefaultWeightedEdge> graph)
+        {
+            return new PrimMinimumSpanningTree<>(graph);
+        }
+
+        @Override
+        public String toString()
+        {
+            return "Prim";
+        }
+    }
+
+    public static class KruskalBenchmark
+        extends BenchmarkBase
+    {
+        @Override
+        MinimumSpanningTreeAlgorithm<DefaultWeightedEdge> createSolver(
+            Graph<Integer, DefaultWeightedEdge> graph)
+        {
+            return new KruskalMinimumSpanningTree<>(graph);
+        }
+
+        @Override
+        public String toString()
+        {
+            return "Kruskal";
+        }
+    }
+
+    public static class BoruvkaBenchmark
+        extends BenchmarkBase
+    {
+        @Override
+        MinimumSpanningTreeAlgorithm<DefaultWeightedEdge> createSolver(
+            Graph<Integer, DefaultWeightedEdge> graph)
+        {
+            return new BoruvkaMinimumSpanningTree<>(graph);
+        }
+
+        @Override
+        public String toString()
+        {
+            return "Boruvka";
+        }
+    }
+
+    public void testBenchmark()
+        throws RunnerException
+    {
+        System.out.println("Minimum Spanning Tree Benchmark");
+        System.out.println("-------------------------------");
+        System.out.println(
+            "Using G(n,p) random graph with n = " + PERF_BENCHMARK_VERTICES_COUNT + ", p = "
+                + PERF_BENCHMARK_EDGES_PROP);
+        System.out.println("Warmup phase " + WARMUP_REPEAT + " executions");
+        System.out.println("Averaging results over " + REPEAT + " executions");
+
+        List<Supplier<BenchmarkBase>> algFactory = new ArrayList<>();
```

Comment:
Out of curiosity, what's the advantage of this supplier/factory approach versus:
BenchmarkBase[] benchmarks={new PrimBenchmark(), new KruskalBenchmark(), new BoruvkaBenchmark()};
for(BenchmarkBase benchmark: benchmarks){...}